server/app.js

```diff
@@ -0,0 +1,75 @@
+import express from 'express';
+import path from 'path';
+import bodyParser from 'body-parser';
+import cookieParser from 'cookie-parser';
+import dotenv from 'dotenv';
+import logger from 'morgan';
+import cors from 'cors';
+import webpack from 'webpack';
+import webpackDevMiddleware from 'webpack-dev-middleware';
+import webpackHotMiddleware from 'webpack-hot-middleware';
+import webpackConfig from '../webpack.config';
+
+// Set up the express app and middleware
+const app = express();
+
+const compiler = webpack(webpackConfig);
+
+const env = process.env.NODE_ENV || 'development';
+console.log('>>>>>> ENV: ', env);
+
+dotenv.config();
+
+if (env === 'development') {
+  dotenv.config();
+  app.use(webpackDevMiddleware(compiler, {
+    historyApiFallback: true,
+    hot: true,
+    noInfo: true,
+    publicPath: webpackConfig.output.publicPath
+  }));
+  app.use(webpackHotMiddleware(compiler, {
+    log: console.log
+  }));
+}
+
+// set morgan to log info about our requests for development use.
+app.use(logger('dev'));
+
+// initialize body-parser to parse incoming parameters requests to req.body
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+
+// initialize cookie-parser to allow us access
+// the cookies stored in the browser.
+app.use(cookieParser());
+
+// Enable Cross-Origin Resource Sharing
+const corsOptions = {
+  allowHeaders: ['Content-Type', 'x-auth'],
+  exposedHeaders: ['x-auth'],
+  // some legacy browsers (IE11, various SmartTVs) choke on 204
+  optionsSuccessStatus: 200
+};
+app.use(cors(corsOptions));
+
+const publicPath = path.join(__dirname, '../client/public/');
+const indexPath = path.resolve(__dirname, publicPath, 'index.html');
+
+app.use('/', express.static(publicPath));
+
+// Require our routes
+require('./routes')(app);
```

Comment:
I will suggest you are consistent your with coding style, I can see you are using es6 module system which is `import/export` but right here you happen to be using the commonjs module system which is `require` here.